src/coreclr/vm/threadsuspend.cpp

```diff
@@ -5744,6 +5769,58 @@ BOOL CheckActivationSafePoint(SIZE_T ip)
     return isActivationSafePoint;
 }
 
+// This function is called when thread suspension is pending for debugger if APC is enabled.
+// As it's not allowed to change the IP while paused in an APC Callback for security reasons 
+// if CET is enabled, the goal of this function is to enable single step in the paused thread
+// and then pause using single step exception after exit the APC callback
+// this will allow the debugger to setIp to execute FuncEvalHijack.
+#ifdef FEATURE_SPECIAL_USER_MODE_APC
+void HandleSuspensionForInterruptedThreadForDebugger(CONTEXT *interruptedContext)
```

Comment:
Instead of cloning the `HandleSuspensionForInterruptedThread`, could you please add a bool argument indicating it is for debugger to the original function and handle it appropriately there?